### DISCUSSION TITLE
Project.repository_merge_base([...]) with forks (remotes)

### DISCUSSION THREAD

**Discussion by zerothi** [python-gitlab/python-gitlab#2883] [Q&A] [answered]
I am trying to streamline some cherry-picking for gitlab workflows.

The principle is simple:
1. Create a merge-request and label it with a specific label that marks it to be backported (cherry picked)
2. Run a script that can create the necassary cherry-pick command

Now, I can rather simple get to the point where I have the source branch and target branch as objects in the API workflow:
```python
# def get_project(project_id) -> Project
mr = project.mergerequests.get(id)
project = get_project(project_id)
source = get_project(mr.source_project_id, mr.manager.gitlab).branches.get(mr.source_branch)
target = project.branches.get(mr.target_branch)
```
Generally these two originates from different sources, i.e. `target.web_url != source.web_url`.

I tried:
```python
mb = project.repository_merge_base([
    target.commit["web_url"],
    source.commit["web_url"]
])
```
to no avail (not that i fully expected this to work)... 

So basically, how can I use `gitlab` to extract the merge-base between the two branches? Or, can I retrieve the commits that will be added (or has been added) to `target` so I can create the necessary cherry-picks?

The merge-request can either be merged, or not merged, both cases are interesting!

**Comment by JohnVillalovos** [comment, marked as answer]
First thing I would do is look at the upstream documentation:

https://docs.gitlab.com/ee/api/repositories.html#merge-base

From my quick glance I don't think it will work how you hope. I believe it only takes `refs` aka SHA values.

One thing that might be helpful for debugging is:
`target.pprint()` that should print out all the values in the object. Which can be helpful. All the Python GitLab objects support `pprint()`, `pformat()`, `to_json()`, and `asdict()`. Depending on the situation they can be useful in debugging code.

**Reply by zerothi** [reply]
Thanks for your feedback, indeed it doesn't look like it. 

As for the debugging, I already played with `pprint` with great satisfaction. ;)
